optuna/integration/allennlp.py

```diff
@@ -21,6 +21,32 @@
     TrackerCallback = object
 
 
+def _save_best_config(config_file: str, study: optuna.Study) -> Dict:
+    best_params = study.best_params
+    for key, value in best_params.items():
+        best_params[key] = str(value)
+    config = json.loads(_jsonnet.evaluate_file(config_file, ext_vars=best_params))
+    return allennlp.common.params.infer_and_cast(config)
+
+
+def save_best_config(input_config_file: str, output_config_file: str, study: optuna.Study) -> None:
+    """Save resulting jsonnet replacing masks with best params in the experiment.
+
+    Args:
+        input_config_file:
+            Config file used in AllenNLPExecutor.
+        output_config_file:
+            Path to output a optimized jsonnet config file.
+        study:
+            A optimized study instance.
+            ``optimized`` means it requires ``study.best_trial_id`` is not empty.
+
+    """
+    best_config = _save_best_config(input_config_file, study)
```

Comment:
If you are happy with my above suggestion of renaming...
```suggestion
    best_config = _dump_best_config(input_config_file, study)
```